contracts/LockedFund/LockedFund.sol

```diff
@@ -502,53 +661,123 @@ contract LockedFund is ILockedFund {
 	/**
 	 * @notice Creates a Vesting Contract for a user.
 	 * @param _tokenOwner The owner of the vesting contract.
+	 * @param _vestingData The vesting details like cliff & duration in short form.
 	 * @return _vestingAddress The Vesting Contract Address.
 	 * @dev Does not do anything if Vesting Contract was already created.
 	 */
-	function _createVesting(address _tokenOwner) internal returns (address _vestingAddress) {
-		require(cliff[msg.sender] != 0 && duration[msg.sender] != 0, "LockedFund: Cliff and/or Duration not set.");
-		/// Here zero is given in place of amount, as amount is not really used in `vestingRegistry.createVesting()`.
-		vestingRegistry.createVesting(_tokenOwner, 0, cliff[_tokenOwner], duration[_tokenOwner]);
-		_vestingAddress = _getVesting(_tokenOwner);
-		emit VestingCreated(msg.sender, _tokenOwner, _vestingAddress);
+	function _createVesting(address _tokenOwner, bytes32 _vestingData) internal returns (address[] memory) {
+		address[] memory _vestingAddresses = new address[](userVestings[_tokenOwner].length);
```

Comment:
if you initialize the array within if/else block, you can save some gas on memory allocation for the else case

Reply:
Solved in 70d811e